src/main.js

```diff
@@ -224,19 +201,74 @@ function addIssDataToDashboard(iss_position, populationSpace, ipstackData) {
     createElementAndAppend({ parentElement: issDataDiv, elementType: 'div', elementId: 'issPopulation', classNames: 'text-xl', textVal: "pop:" + populationSpace });
 }
 
-function compareGini(x) {
-    if (x > 50) {
-        return 'very high';
-    } else if (x > 40) {
-        return 'high';
-    } else if (x > 30) {
-        return 'medium';
-    } else return 'low';
+class ColorStep {
+    constructor(time, color) {
+        this.time = time;  // Number of minutes since 0:00
+        this.color = color;  // {r: red, g: green, b: blue}
+    }
+}
+
+function linearInterpolation(start, end, weight) {
+    return start + (end - start) * weight;
+}
```

Comment:
I think what we're doing here is typically termed "tweening":

https://en.wikipedia.org/wiki/Inbetweening

And this looks like a easing function. There are many available: https://easings.net/

One way they can be used is in a function like this:

```js
export function tween(start, end, duration, ease = easeInOutSine) {
  const startTime = Date.now();
  const delta = end - start;

  return () => {
    const elapsed = Date.now() - startTime;
    if (elapsed >= duration) {
      return end;
    } else {
      const fraction = easeInOutSine(elapsed / duration);
      return delta * fraction + start;
    }
  };
}

function easeInOutSine(x: number): number {
  return -(Math.cos(Math.PI * x) - 1) / 2;
}
```

Where everytime you call the function that `tween` returns, you get the value at that point in time.

I think you've done fine here, but just showing you this reasonably common way of structuring these things in case you're interested, and you might see the term "tween" in your JS travels

Reply:
Cool!